scalding-core/src/main/scala/com/twitter/scalding/ExecutionContext.scala

```diff
@@ -31,6 +36,38 @@ trait ExecutionContext {
   def flowDef: FlowDef
   def mode: Mode
 
+  private def getIdentifierOpt(descriptions: Seq[String]): Option[String] = {
+    if (descriptions.nonEmpty) Some(descriptions.distinct.mkString(", ")) else None
+  }
+
+  private def updateJobName(step: BaseFlowStep[JobConf], descriptions: Seq[String]): Unit = {
+    val conf = step.getConfig
+    getIdentifierOpt(getDesc(step)).foreach(id => {
+      val newJobName = "%s %s".format(conf.getJobName, id)
+      println("Added descriptions to job name: %s".format(newJobName))
```

Comment:
I already pushed a change that would put the descriptions in a different config and not update mapred.name.  Since we want it appended to the name I will add a steplistener on our side that will just add the value to that config to the job name on our side... win win for everyone :)